### ISSUE TITLE
CSS Variables break the local server.

### ISSUE THREAD

**Issue by lukethacoder** [forcedotcom/lwc-dev-server-feedback#76]
**Describe the bug**
When using Web Standard CSS variables, the local dev server breaks.

```css
:host {
  --brand-primary: red;
}
```

**To Reproduce**
1. open any LWC
2. add a CSS Variable to any selector
4. see error

**Expected behavior**
It should not break the localdev server.

**Screenshots**

![image](https://user-images.githubusercontent.com/13529535/93843860-b4886b80-fcde-11ea-87bc-1a146f57c499.png)

screenshot of the error thrown when attempting to use CSS Variables

**Desktop:**
- OS: Windows 10 (2004 - 19042.508)
- Browser: Edge (85.0.564.51) & Firefox Developer Edition (77.0b9)
- Node 12.13.0
- sfdx-cli/7.74.1 win32-x64 
- @salesforce/lwc-dev-server 2.5.1

**Additional context**

Seems almost related to https://github.com/forcedotcom/lwc-dev-server-feedback/issues/40 , but this is for standard css variables, not Salesforce Design Tokens.

**Comment by mad-magic** [comment]
Can confirm - ran across same issue last week:
- @salesforce/lwc-dev-server 2.5.1
- Browser: Chrome (87.0.4280.88)
- sfdx-cli/7.80.0-b979aae708 win32-x64 node-v12.18.3

**Comment by samuelcarreira-ibs** [comment]
The bug persist since the API 49 (when I've started develop LWCs)!
You can deploy to the org and it works, but you cannot use the `:host` CSS selector to target your host

- sfdx-cli/7.81.0-7b953c80d1 win32-x64 node-v12.18.3
- Edge Version 87.0.664.55 (Official build) (64-bit)

**Comment by AllanOricil** [comment]
This bug persists, but now it displays a different message.

![image](https://user-images.githubusercontent.com/55927613/134044775-c001204a-35d5-4cb9-b096-fd958d64ba93.png)


Steps to reproduce:

declare any css host or root variable, like

````css
:host {
    --simple-text-color: #aaaaaa;
    --strong-text-color: #585858;
}
````

start the dev server, open the component and verify the error message from the image above.

Acording to the docs, LWC does not allow root variables in the component level, so it is expected to display an error message. But host variables are possible, and they work inside a Salesforce org. So this error message should not happen in the dev server when we declare host variables

**Comment by impaler** [comment]
I also just encountered this issue when following the salesforce Developer Guide "Create Styling Hooks for Your Components"

https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.create_components_css_styling_hooks

Eg

```
:host {
    --important-color: red;
}

.important {
    color: var(--important-color);
}
```

And got the error message `"Couldn't find the compiled component. If this component has a dependency on a component in the org or a component in a package in the org, test this component directly in the org."`

The error is only present when using the css variables syntax.

It would be really nice to have support for this since its the recommended way in the docs to expose styling hooks to the components. Thank you.

**Comment by apop880** [comment]
Seeing this as well. Is there any workaround which will allow me to use CSS variables across my project but still test locally?
![image](https://user-images.githubusercontent.com/41379678/154148547-2ac910ab-239e-4f63-b1f5-7de1d5c36fe0.png)

**Comment by vilchesalves** [comment]
How is this bug a p3?

It basically makes styling hooks as defined in the lwc guide unusable on a professional environment.

**Comment by lukethacoder** [comment]
Is there any update on this? Would love to be using the local dev server but I'm blocked by this issue not being resolved.

**Comment by andruts** [comment]
Faced this issue today.

```css
:host {
  --important-color: red;
}
```

CssSyntaxError: LWC1009: postcss-plugin-lwc: 2:3: Invalid definition of custom property "--important-color".

sfdx-cli/7.148.3 win32-x64 node-v14.17.1

**Comment by dslagman** [comment]
Experiencing the same issue: `Uncaught SyntaxError: Unexpected token ':'`

```
:host {
    --lwc-spacingSmall : 0.4rem;
}
```

**Comment by mmorozov-noltic** [comment]
Hey, how can we fix this problems? Any ideas?

**Comment by vilchesalves** [comment]
Sorry for the low effort comment, but "bump".

**Comment by attilah** [comment]
The root of this cause is that dev server depends on an older version of style compiler which HAD this restriction. But this restriction was removed in [this](https://github.com/salesforce/lwc/commit/6a41fb82cda8641751f65d31c0456e741a6280c1) commit, so if Salesforce would update the LWC (and other) dependencies and release a new version of this dev server this specific problem would go away.

**Comment by mmorozov-noltic** [comment]
wow wow. Is is very good news. Thank you! I will wait this release!

**Comment by lukethacoder** [comment]
Any timelines on release? That PR was merged in a year ago? 

Ashame this isn't open source, could allow the community to fork and keep it alive and working.

**Comment by attilah** [comment]
Not to their defense, but they are using a lot of not yet open sourced package so open sourcing the devserver would not be a small task, but it would be a great addition. My biggest concern on why they are not on it is the npm download counts...the dev server and other package stats are just so low...but of course it is a chicken-egg problem...

**Comment by vilchesalves** [comment]
Sorry for the low effort comment, but "bump".

**Comment by balindersingh** [comment]
I am seeing same issue
` CssSyntaxError: LWC1009: postcss-plugin-lwc: 3:5: Invalid definition of custom property ":--card-color"`

**Comment by erocheleau** [comment]
Same issue here: 

`CssSyntaxError: LWC1009: postcss-plugin-lwc: 2:3: Invalid definition of custom property "--slds-c-checkbox-color-border".`

When trying to edit :host variables that are listed under the styling hooks in the lightning design system documentation.

```
:host {
  --slds-c-checkbox-color-border: var(--lwc-colorTextIconDefault, #747474);
  --slds-c-checkbox-color-border-checked: var(--lwc-colorTextIconDefault, #747474);
}
```

**Comment by jsmithdev** [comment]
Dang, over 2 years... I guess it'll be stale in 11 days and ignored but same issue with not being able to use CSS as documented by Salesforce. System is up to date. 

`CssSyntaxError: LWC1009: postcss-plugin-lwc: 2:5: Invalid definition of custom property`

**Comment by erocheleau** [comment]
@jsmithdev Yea we've had to include a step in our readme to tell people who wanted to develop using the local dev server to "comment-out" the styling hooks usage...

**Comment by benjaminloerincz** [comment]
"I guess it'll be auto marked stale again soon" :-(

**Comment by CavMejia** [comment]
a pitty..... hope they put some more effort on this tool, really deserves it

**Comment by nstuyvesant** [comment]
So does this mean we can't use CSS custom properties in LWC or just that we can't test anything that uses it in the LWC development server?

In my case, I was not using :root - just a local class.
```css
.spai-scores {
  --spai-stroke-width: 0.875rem;
  --spai-score-font-weight: 600;
  --spai-incomplete-stroke: #edeef4;
  --spai-dashed-stroke-color: white;

  --spai-major-score-diameter: 9.125rem;
  --spai-major-font-size: 2.875rem;

  --spai-minor-score-diameter: 5.938rem;
  --spai-minor-font-size: 1.5rem;
}
```

**Comment by lukethacoder** [comment]
> So does this mean we can't use CSS custom properties in LWC or just that we can't test anything that uses it in the LWC development server?

You can still use CSS variables; it just means you forfeit the local dev server

**Comment by todthomson** [comment]
Can someone from Salesforce please fix this bug (or release `lwc-dev-server` as open source software so we can fix it)?

**Comment by todthomson** [comment]
> It would be really nice to have support for this since its the recommended way in the docs to expose styling hooks to the components. Thank you.

Indeed, the recommended (and only reasonable) way to style LWCs (Lightning Web Components) is to override the CSS variables in, e.g.

```
:host
{
    --slds-c-accordion-sizing-border: 0;
}
```
or a css class attached (for tighter scoping), e.g.

```
.accordion
{
    --slds-c-accordion-summary-color-background: #f5f5f5;
}
```

So, in my opinion, this bug is a P1, because it means you cannot use the LWC Dev Server in combination with the recommended way to style LWCs.

It does not give me any confidence to recommend Salesforce (in particular LWC development) when such a fundamental bug has gone unresolved for **more than three years.**

Please do the needful here and fix this bug.

Better yet, assign some development resources to at least fixing bugs in this offering (or remove it from your documentation, so as to not waste people's time like this).

Even better, why not open source this so that people can fix these bugs themselves?

**Comment by lukethacoder** [comment]
After waiting about 4 years (and then seeing we won't be getting anything [until at least 2025](https://github.com/forcedotcom/salesforcedx-vscode/issues/5501)) I decided to build something myself over a few weekend or two.

- https://lwc.garden/
- https://github.com/lukethacoder/lwc-garden
- https://stackblitz.com/edit/lwc-garden-minimal-config?file=garden.config.js

Full support for CSS Variables and anything else you could want from the modern web.

- 🚀 Local First Development (feat. Speedy HMR)
- 🧪 Mock Anything (Apex, OmniScript, 3rd Party Packages and on-platform imports)
- 🥧 `<slot/>` Placeholders
- 💅 SLDS Support OOTB (and `lightning-base-components`)
- ✏️ API Editor via Storybook-like argTypes
- 🎨 Custom Theming API
- ⚙️ Config as much or as little as your like
- 🙌 And best of all, open sourced 

![lwc-garden-arg-types](https://github.com/forcedotcom/lwc-dev-server-feedback/assets/13529535/6dfa9e87-8a62-41fe-8f64-642001b7056b)

![lwc-garden-lighting-base-components](https://github.com/forcedotcom/lwc-dev-server-feedback/assets/13529535/6a9da02d-1433-47bf-894b-6245df132a38)
